borg/archiver.py

```diff
@@ -607,8 +607,27 @@ def do_break_lock(self, args):
         start to end, or must match from the start to just before a path separator.
         Except for the root path, paths will never end in the path separator when
         matching is attempted.  Thus, if a given pattern ends in a path separator, a
-        '*' is appended before matching is attempted.  Patterns with wildcards should
-        be quoted to protect them from shell expansion.
+        '*' is appended before matching is attempted.
+
+        Regular expressions similar to those found in Perl are also supported. Unlike
+        shell patterns regular expressions are not required to match the complete
+        path and any substring match is sufficient. It is therefore strongly
+        recommended to anchor patterns to the start ('^'), to the end ('$') or both.
+        Path separators ('\\' for Windows and '/' on other systems) in paths always
+        normalized to a forward slash ('/') before applying a pattern. The regular
+        expression syntax is described in the `Python documentation for the re module
+        <https://docs.python.org/3/library/re.html>`_.
+
+        Exclusions can be passed via command line options. The `--exclude` option
+        (`-e` for short) uses shell pattern syntax and `--exclude-regex` uses regular
+        expression syntax. When passed via a shell the patterns should be quoted to
+        protect them from expansion.
+
+        The `--exclude-from` option permits loading exclusion patterns from a text
+        file with one pattern per line. Empty lines as well as lines starting with
+        the number sign ('#') are ignored. If a line starts with '{RE}' (case
```

Comment:
the point is that if we introduce --exclude-regex now then we would need to have to introduce some --exclude-glob soon later (that implements the modified fnmatch behaviour).

i don't want that many options (especially since with a file, it also comes out with 1 option), thus the solution would be to always use --exclude and allow a prefix.

if we default to fn: (or rather fm:) now (which is compatible to previous behaviour, but obviously using the strange fnmatch stdlib matching then), we can change the default to gl: at 1.0 (which then would behave more like people expect it to behave).

btw, in general please use lowercase prefixes for code and documentation. having to type all with shift is more work, esp. in case you have a lot of patterns.